mlflow/utils/autologging_utils.py

```diff
@@ -306,6 +310,80 @@ def batch_metrics_logger(run_id):
     batch_metrics_logger.flush()
 
 
+def _check_version_in_range(ver, min_ver, max_ver):
+    return LooseVersion(min_ver) <= LooseVersion(ver) <= LooseVersion(max_ver)
+
+
+def _load_version_file_json():
+    version_file_path = resource_filename(__name__, "../ml-package-versions.yml")
+    with open(version_file_path) as f:
+        return yaml.load(f, Loader=yaml.SafeLoader)
+
+
+_version_file_json = _load_version_file_json()
+
+
+# A map FLAVOR_NAME -> list of dependent (module_name, key_in_version_file_json)
+_cross_tested_flavor_to_module_name_and_json_key = {
+    "fastai": [("fastai", "fastai-1.x")],
+    "gluon": [("mxnet", "gluon")],
+    "keras": [("keras", "keras")],
+    "lightgbm": [("lightgbm", "lightgbm")],
+    "statsmodels": [("statsmodels", "statsmodels")],
+    "tensorflow": [("tensorflow", "tensorflow")],
+    "xgboost": [("xgboost", "xgboost")],
+    "sklearn": [("sklearn", "sklearn")],
+    "pytorch": [("torch", "pytorch"), ("pytorch_lightning", "pytorch-lightning")],
+}
+
+
+def _get_min_max_version_and_pip_release(json_key):
+    min_version = _version_file_json[json_key]["autologging"]["minimum"]
+    max_version = _version_file_json[json_key]["autologging"]["maximum"]
+    pip_release = _version_file_json[json_key]["package_info"]["pip_release"]
+    return min_version, max_version, pip_release
+
+
+def _is_autologging_supported(flavor_name, get_module_version_fn):
+    def _check_autologging_supported(module_name, json_key):
+        min_version, max_version, _ = _get_min_max_version_and_pip_release(json_key)
+        return _check_version_in_range(get_module_version_fn(module_name), min_version, max_version)
+
+    return all(
+        [
+            _check_autologging_supported(module_name, json_key)
+            for module_name, json_key in _cross_tested_flavor_to_module_name_and_json_key[flavor_name]
+        ]
+    )
+
+
+def is_autologging_supported(flavor_name):
+    def get_module_version(module_name):
+        return importlib.import_module(module_name).__version__
+
+    return _is_autologging_supported(flavor_name, get_module_version)
+
+
+def gen_autologging_package_version_requirements_doc(flavor_name):
+    def _gen_single_requirement(json_key):
+        min_ver, max_ver, pip_release = _get_min_max_version_and_pip_release(json_key)
+        return "{min_ver}<={pip_release}<={max_ver}".format(
```

Comment:
Formatting suggestion to render prettier / more readable rst:

```suggestion
        return "``{min_ver}`` <= ``{pip_release}`` <= ``{max_ver}``".format(
```